rhcos4/profiles/moderate.profile

```diff
@@ -548,7 +548,7 @@ selections:
 
     # AU-3
     - package_audit_installed
-    - grub2_audit_argument
+    - bls_audit_option
```

Comment:
The way how to check BLS entry options may be similar across OSes or arches, but description and OCIL can vary a lot. Take for example [`zipl_audit_argument`](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/bootloader-zipl/zipl_audit_argument/rule.yml), description and OCIL mention a few commands and config files specific for zipl.

So I wonder if the rule should be renamed to `grub2_bls_audit_option`.
And conversely, `zipl_audit_argument` renamed to `zipl_bls_audit_option`.

The template name can remain boot loader neutral, but I would put in singular: `bls_entry_option`.
@evgenyz Thoughts?

Reply:
Entries are plural because we are checking ALL entries: `<ind:filepath operation="pattern match">^/boot/loader/entries/.*\.conf$</ind:filepath>`.

Reply:
oh, wouldn't it be the case that this won't pass even after the remediation is applied? The deployment keeps the previous configuration around which wouldn't have the compliant option.

Reply:
Now, `zipl_audit_argument` (as it seems to me) holds 2 different *unrelated* checks: for option in BLS-compatible entry and for settings in /etc/zipl.conf. I thinks that they should be separated (`bls_audit_option` + `zipl_config_blah_blah`). The `bls_audit_option` is bootloader-agnostic and could be reused.

Reply:
> oh, wouldn't it be the case that this won't pass even after the remediation is applied? The deployment keeps the previous configuration around which wouldn't have the compliant option.

Yes, it would be the case.

Reply:
why make it so strict? it would make it almost impossible for customers to get to a compliant state.

Reply:
I think we discussed this with Gabe some time ago and reached a compromise where it would have been enough to check the "current" bootloader settings plus the running kernel via `/proc/cpuinfo`. We can still follow up with either BLS or rpm-ostree developers to introduce a mode to replace all entries, but it shouldn't be the default anyway because you /want/ to be able to go back to the configuration before the changes were applied in case the node wouldn't boot up properly with the additional kargs.

Reply:
Basically because of: https://github.com/ComplianceAsCode/content/issues/5285#issuecomment-617363140

But, also, this rule is a generic BLS-compatible check, it does not take into account configuration shenanigans of CoreOS.

Reply:
Yet!

Reply:
Later we could add something like `current=True` parameter to this template.

Reply:
Later, this template can also be enhanced to consider cases when `$kernelopts` is present in the entry.

Reply:
> Later, this template can also be enhanced to consider cases when `$kernelopts` is present in the entry.

I would be careful with that, it will make the rule not compatible with BLS.

Reply:
But, on the other hand we can parametrize the template to create `grub2_bls_audit_option`.

Reply:
The template can have conditionals for how different products handle BLS.
RHEL for example may check for all entries, not just the current.